### PR TITLE
[jit] Improve documentation around builtin functions

### PR DESCRIPTION
**Pull request by driazati**
This breaks the builtins page into some more sections and adds details about Python built-in functions

Differential Revision: [D18718166](https://our.internmc.facebook.com/intern/diff/18718166/)

### CODE REVIEW

**Comment by eellison** [comment]
Can you post a before and after please ?

**Comment by driazati** [comment]
You can see it here https://driazati.github.io/pytorch_doc_previews/jit.html#built-in-functions-and-modules

**Review by eellison — COMMENTED** [review]
i don't see anything wrong, it would still be helpful to post one before & after of how the builtins / tensor methods rendering changed

**Comment by driazati** [comment]
This is the before: https://pytorch.org/docs/master/jit_builtin_functions.html#python-built-in-functions
And after: https://driazati.github.io/pytorch_doc_previews/jit_builtin_functions.html#python-built-in-functions

File: torch/jit/supported_ops.py [hunk 1]
```diff
@@ -58,70 +83,210 @@ def hidden(name):
...
+        'range': 'fake::does_not_exist',
+    }
+
+    schemaless_op_explanations = {
+        'print': 'Print any value',
+        'tuple': 'Lists cannot be converted to tuples with this method since their size is not statically known',
+        'getattr': 'Attribute name must be a literal string',
+        'hasattr': 'Attribute name must be a literal string',
+        'isinstance': 'Result is static',
+        'zip': 'Arguments must be lists',
```
Comment (eellison) [inline torch/jit/supported_ops.py]: This isn't quite true. We can iterate over Tensors, Lists, Tuples, Dictionaries, Strings, and ModuleLists / ModuleDicts. We should probably have a note on iterables in torchscript, which you then reference here.

**Review by eellison — APPROVED** [review]
Generally looks good. Supported Tensor and Pytorch Methods is incredibly long. It would be nice if those were collapsable. 

Also, the `math` module is including ops with tensor in them. You could add a filter for math module so that we don't accidentally include these ops.

`math.ceil(self : Tensor,
          out : Tensor) -> Tensor`

The iterator thing isn't exactly right. I should probably add a section in JIT docs about how we handle them.

File: torch/jit/supported_ops.py [hunk 2]
```diff
@@ -203,7 +203,8 @@ def _get_global_builtins():
         'getattr': 'Attribute name must be a literal string',
         'hasattr': 'Attribute name must be a literal string',
         'isinstance': 'Result is static',
-        'zip': 'Arguments must be lists',
+        'zip': 'Arguments must be a Tensor, list, tuple, dictionary, '
+               ':any:`torch.nn.ModuleList` or :any:`torch.nn.ModuleDict`',
         'enumerate': 'Arguments must be lists',
```
Comment (eellison) [inline torch/jit/supported_ops.py]: same thing for zip applies to enumerate, also you forgot string here

Comment (driazati) [reply torch/jit/supported_ops.py]: I added an iterables section, we can expand on it in a follow up

**Pull request closed without merging**